src/controllers/hid/hidioglobaloutputreportfifo.cpp

```diff
@@ -0,0 +1,97 @@
+#include "controllers/hid/hidioglobaloutputreportfifo.h"
+
+#include <hidapi.h>
+
+#include "controllers/defs_controllers.h"
+#include "controllers/hid/legacyhidcontrollermappingfilehandler.h"
+#include "util/compatibility/qbytearray.h"
+#include "util/compatibility/qmutex.h"
+#include "util/string.h"
+#include "util/time.h"
+#include "util/trace.h"
+
+namespace {
+constexpr int kMaxHidErrorMessageSize = 512;
+constexpr int kSizeOfFifoInReports = 32;
+} // namespace
+
+HidIoGlobalOutputReportFifo::HidIoGlobalOutputReportFifo()
+        : m_fifoQueue(kSizeOfFifoInReports) {
+}
+
+void HidIoGlobalOutputReportFifo::addReportDatasetToFifo(const quint8 reportId,
+        const QByteArray& reportData,
+        const mixxx::hid::DeviceInfo& deviceInfo,
+        const RuntimeLoggingCategory& logOutput) {
+    // First byte must always be the ReportID-Byte
+    QByteArray report(std::move(reportData));
+    report.prepend(reportId);
```

Comment:
Writing the first byte and then appending the remaining data would be more efficient.

Using `std::move` on `const QByteArray&` is pointless.

Reply:
> Writing the first byte and then appending the remaining data would be more efficient.

This was the old code, I changed it because the Qt6 documentation of prepend says: "This operation is typically very fast ([constant time](https://doc.qt.io/qt-6/containers.html#constant-time)), because [QByteArray](https://doc.qt.io/qt-6/qbytearray.html) preallocates extra space at the beginning of the data, so it can grow without reallocating the entire array each time."

Reply:
Ok. Maybe a short comment/link would help? If you are not a Qt expert those subtle details are surprising.